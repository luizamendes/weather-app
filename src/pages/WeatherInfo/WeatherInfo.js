import React, { useEffect, useState } from "react";
import { Button } from "../../components/Button";
import "./index.css";

export const WeatherInfo = () => {
  const [place, setPlace] = useState("");
  const [maxTemp, setMaxTemp] = useState(0);
  const [minTemp, setMinTemp] = useState(0);
  const [currentTemp, setCurrentTemp] = useState(0);

  const queryString = window.location.search;

  useEffect(() => {
    const urlParams = new URLSearchParams(queryString);
    setMaxTemp(urlParams.get("max"));
    setMinTemp(urlParams.get("min"));
    setPlace(urlParams.get("place"));
    setCurrentTemp(urlParams.get("current"));
  }, [queryString]);

  return (
    <div className="weather-info">
      <div className="weather-info__content">
        <h2>Temperature</h2>
        <h3>{place}</h3>
        <p>
          <spam>Current temperature: </spam>
          {currentTemp} °C
        </p>
        <p>
          <spam>Maximum temperature: </spam>
          {maxTemp} °C
        </p>
        <p>
          <spam>Minimum temperature: </spam>
          {minTemp} °C
        </p>
        <Button onClick={() => window.close()}>Return to map</Button>
      </div>
    </div>
  );
};
