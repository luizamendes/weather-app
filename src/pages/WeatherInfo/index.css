.weather-info {
  display: flex;
  height: 100vh;
  justify-content: center;
  align-items: center;
  background: #e7e7e7;
}

.weather-info__content {
  display: flex;
  background-color: #fff;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 3rem 6rem;
  border-left: 3px solid #ff8c00;
  box-shadow: 10px 10px 20px 0px rgba(0, 0, 0, 0.75);
}

.weather-info__content h2 {
  margin-bottom: 0;
}

.weather-info__content p {
  margin: 0.5rem 0;
}

.weather-info__content p spam {
  font-weight: bold;
}
