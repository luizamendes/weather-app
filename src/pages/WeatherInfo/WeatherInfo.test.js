import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { WeatherInfo } from "./WeatherInfo";

describe("WeatherInfo page Snapshot", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <WeatherInfo />
        </Router>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
