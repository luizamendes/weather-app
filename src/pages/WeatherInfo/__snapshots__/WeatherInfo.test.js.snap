// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherInfo page Snapshot renders correctly 1`] = `
<div
  className="weather-info"
>
  <div
    className="weather-info__content"
  >
    <h2>
      Temperature
    </h2>
    <h3>
      
    </h3>
    <p>
      <spam>
        Current temperature: 
      </spam>
      0
      °C
    </p>
    <p>
      <spam>
        Maximum temperature: 
      </spam>
      0
      °C
    </p>
    <p>
      <spam>
        Minimum temperature: 
      </spam>
      0
      °C
    </p>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      Return to map
    </button>
  </div>
</div>
`;
